iroh/src/endpoint/connection.rs

```diff
@@ -0,0 +1,1415 @@
+//! The [`Connection`] wraps a `quinn::Connection`.
+//!
+//! The [`Connection`] is how you send data to and receive data from the remote endpoint.
+//!
+//! There are many transitions states between attempting to start a connection and
+//! receiving a cryptographically secure connection.
+//!
+//! The main items in this module are:
+//!
+//! - [`Connection`] to create streams to talk to a remote endpoint.
+//! - [`Connecting`] to intercept a connection before it is fully formed.
```

Comment:
```suggestion
//! - [`Connecting`] for operating on connections that haven't finished their handshake yet.
```

I feel like "intercept" might give the wrong impression (think intercepting "foreign" connections).